### PR TITLE
microbit: Added getDigitalValue overload for PullMode #156

### PR DESCRIPTION
**Pull request by finneyj**
- Introduced an overload to MicroBitIOPin::getDigitalValue() to permit the
setting of a specific pull mode at the time of reading.

 - Bugfix of MicroBitIOPin::setPull() to persist preferred pull settings

 - Added configuration options to allow the default PullMode to be set via
   compile time option through MicroBitConfig.h or YOTTA_CONFIG

### CODE REVIEW

File: inc/drivers/MicroBitPin.h [hunk 1]
```diff
@@ -95,8 +95,8 @@ class MicroBitPin : public MicroBitComponent
 {
     // The mbed object looking after this pin at any point in time (untyped due to dynamic behaviour).
     void *pin;
-
     PinCapability capability;
+    uint8_t pullMode;
```
Comment (dpgeorge) [inline inc/drivers/MicroBitPin.h]: This is going to add (probably) 4 bytes of RAM to each pin.  That means taking about 80 bytes when all ~20 pins are created.

Comment (finneyj) [inline inc/drivers/MicroBitPin.h]: checked that - fits within current word alignment bounds, so adds no additional overhead.

Comment (dpgeorge) [inline inc/drivers/MicroBitPin.h]: Phew!

Comment (finneyj) [inline inc/drivers/MicroBitPin.h]: :-)

**Comment by finneyj** [comment]
Merging early as @pelikhan has kindly agreed to test on PXT if we can get it into master as part of this week's planned tagged interim release. Any further comments will still be welcomed and considered.

**Pull request merged**